### PR TITLE
rusk: Fix transaction encoding over the network

### LINKED ISSUE #526: Transactions are not propagated correctly to the local node

### PR DESCRIPTION
**Pull request by herr-seppia**
Edit transaction tests to check that a local peer receive the correct message

Resolves #526

### CODE REVIEW

File: rusk/src/lib/services/network.rs [hunk 1]
```diff
@@ -187,13 +193,16 @@ mod serialization {
...
     const TX_CATEGORY: u8 = 10;
 
-    const RESERVED_FIELDS_BYTES: [u8; 8] = [0; 8];
-    const TX_HEADER_LEN: u64 = {
-        4u64 + // MAGIC
-        8u64 + // RESERVED_FIELDS
-        4u64 // CHECKSUM
+    const RESERVER_FIELDS_LEN: usize = 8;
+    const RESERVED_FIELDS_BYTES: [u8; RESERVER_FIELDS_LEN] =
+        [0; RESERVER_FIELDS_LEN];
```
Comment (autholykos) [inline rusk/src/lib/services/network.rs:198]: s/RESERVER/RESERVED/g

File: rusk/src/lib/services/network.rs [hunk 2]
```diff
@@ -187,13 +193,16 @@ mod serialization {
...
+    const TX_HEADER_LEN: usize = {
+        u32::SIZE + // MAGIC
+        RESERVER_FIELDS_LEN + // RESERVED_FIELDS
+        32 // CHECKSUM
```
Comment (autholykos) [inline rusk/src/lib/services/network.rs:202]: 32 bytes checksum sound quite exaggerated. You sure of this? Or is this a weirdly named hash?

**Review by autholykos — CHANGES_REQUESTED** [review]
See comments. The hash/CRC should be 4 bytes instead of 32

File: rusk/src/lib/services/network.rs (hunk 2 above)
Comment (herr-seppia) [reply rusk/src/lib/services/network.rs]: It was wrong, the checksum is just the first 4 bytes of the hash

**Review by autholykos — APPROVED** [review]
LGTM

**Pull request merged**